plugin/evm/vm_test.go

```diff
@@ -770,44 +742,343 @@ func TestIssueAtomicTxs(t *testing.T) {
+	require.Equal(indexedExportTx.ID(), exportTx.ID(), "expected ID of indexed import tx to match original txID")
+
+	// getAtomicTx should check the processing blocks for any atomic txs.
+	exportTxFetched, status, height, err := vm.getAtomicTx(exportTx2.ID())
+	require.NoError(err)
+	require.Equal(Processing, status)
+	require.Equal(uint64(3), height, "expected height of export tx to be 3")
+	require.Equal(exportTxFetched.ID(), exportTx2.ID(), "expected ID of fetched export tx to match original txID")
+}
+
+func TestGetAtomicTxFromProcessingBlocks(t *testing.T) {
+	require := require.New(t)
+	importAmount := uint64(1000000000)
+	issuer, vm, _, sharedMemory, _ := GenesisVMWithUTXOs(t, true, genesisJSONApricotPhase3, "", "", map[ids.ShortID]uint64{
+		testShortIDAddrs[0]: importAmount,
+		testShortIDAddrs[1]: importAmount,
+	})
+
+	defer func() {
+		err := vm.Shutdown(context.Background())
+		require.NoError(err)
+	}()
+
+	newTxPoolHeadChan := make(chan core.NewTxPoolReorgEvent, 1)
+	vm.txPool.SubscribeNewReorgEvent(newTxPoolHeadChan)
+
+	// Create a block that has atomic txs and accept it so we can fund our eth address
+	importTx, err := vm.newImportTx(vm.ctx.XChainID, testEthAddrs[0], initialBaseFee, []*secp256k1.PrivateKey{testKeys[0]})
+	require.NoError(err)
+
+	err = vm.issueTx(importTx, true)
+	require.NoError(err)
+
+	<-issuer
+
+	blk, err := vm.BuildBlock(context.Background())
+	require.NoError(err)
+
+	atomicTxBlock := blk.(*chain.BlockWrapper).Block.(*Block)
+
+	err = blk.Verify(context.Background())
+	require.NoError(err)
+
+	require.Equal(blk.Status(), choices.Processing, "Expected status to match")
+
+	err = vm.SetPreference(context.Background(), blk.ID())
+	require.NoError(err)
+
+	err = blk.Accept(context.Background())
+	require.NoError(err)
+
+	require.Equal(blk.Status(), choices.Accepted, "Expected status to match")
+
+	lastAcceptedID, err := vm.LastAccepted(context.Background())
+	require.NoError(err)
+	require.Equal(lastAcceptedID, blk.ID(), "Expected last accepted blockID to be the accepted block")
+
+	newHead := <-newTxPoolHeadChan
+	require.Equal(newHead.Head.Hash(), common.Hash(blk.ID()), "Expected new block to match")
+
+	// Create a block that has no atomic txs
+	tx := types.NewTransaction(0, testEthAddrs[0], big.NewInt(0), 21000, big.NewInt(params.LaunchMinGasPrice), nil)
+	signedTx, err := types.SignTx(tx, types.NewEIP155Signer(vm.chainID), testKeys[0].ToECDSA())
+	require.NoError(err)
+
+	errs := vm.txPool.AddRemotesSync([]*types.Transaction{signedTx})
+	for _, err := range errs {
+		require.NoError(err)
 	}
 
-	if err := blk2.Accept(context.Background()); err != nil {
-		t.Fatal(err)
+	<-issuer
+
+	blk2, err := vm.BuildBlock(context.Background())
+	require.NoError(err)
+
+	noAtomicTxBlock := blk2.(*chain.BlockWrapper).Block.(*Block)
+
+	// All new blocks will be post apricot5
+	vm.chainConfig.ApricotPhase5BlockTimestamp = utils.NewUint64(0)
```

Comment:
It's invalid to modify the rules after we've already started accepting blocks. Can we switch to adding a single test that checks whether after attempting to issue a transaction that is already included in a processing block, the status of that transaction shows up as processing correctly?

Reply:
I see. Should I have two separate tests for pre/post apricot 5? I know you mentioned that we don't need to necessarily support pre apricot 5. I think its worth to test all different input cases here.